lib/datadog/tracing/event.rb

```diff
@@ -73,11 +59,11 @@ def unsubscribe_all!
       end
 
       def publish(*args)
-        subscriptions.each do |key, block|
+        subscriptions.each do |block|
           begin
             block.call(*args)
           rescue StandardError => e
-            Datadog.logger.debug { "Error while handling '#{key}' for '#{name}' event: #{e.message}" }
+            Datadog.logger.debug { "Error while handling '#{name}' event with '#{block}': #{e.message}" }
```

Comment:
Minor: if it doesn't come with any perf cost, can we do `block`'s source location? Would probably help pinpoint the callback that triggered the error.

Reply:
`proc.to_s` has that information, so it would be captured in the output.
```ruby
[3] pry(main)> block = proc {}
[4] pry(main)> puts "#{block}"
#<Proc:0x00007f9b89129540 (pry):3>
```
One thing missing is the backtrace of the exception, I think that would be helpful, let me put that one in, like we do in other places.

Reply:
It looks like this:
```
Custom on_error handler #<Proc:0x00007fbb37579f00 /Users/marco.costa/work/dd-trace-rb/lib/datadog/tracing/span_operation.rb:394> failed, using fallback behavior. Error: I also raise an error. Location: /Users/marco.costa/work/dd-trace-rb/spec/datadog/tracing/tracer_spec.rb:509:in `block (8 levels) in <top (required)>'
```